src/main.ts

```diff
@@ -15,327 +15,406 @@ const deleteTempFileTimeout = 60_000;
-              .setIcon("arrow-up-right")
-              .setTitle(Platform.isMacOS ? "Reveal in Finder" : "Show in system explorer")
-              .onClick(() => {
-                this.app.showInFolder(relativePath);
-              })
-            );
-            menu.addItem((item: MenuItem) => item
-              .setIcon("folder")
-              .setTitle("Reveal file in navigation")
-              .onClick(() => {
-                const file = this.app.vault.getFileByPath(relativePath);
-                if (!file) {
-                  console.warn(`getFileByPath returned null for ${relativePath}`)
-                  return;
-                }
-                this.app.internalPlugins.getEnabledPluginById("file-explorer")?.revealInFolder(file);
-              })
-            );
-          }
-        }
-        break;
-      default:
-        new Notice(`no handler for ${protocol} protocol`);
-        return;
-    }
-
-    this.registerEscapeButton(menu);
-    menu.showAtPosition({ x: event.pageX, y: event.pageY });
-    this.app.workspace.trigger("copy-url-in-preview:contextmenu", menu);
-  }
-
-  onImageMouseUp(event: MouseEvent) {
-    const middleButtonNumber = 1;
-    if (event.button == middleButtonNumber && this.settings.middleClickNewTab) {
-      openImageFromMouseEvent(event, this.app);
-    }
-  }
+	longTapTimeoutId?: number;
+	openPdfMenu?: Menu;
+	preventReopenPdfMenu: boolean;
+	lastHoveredLinkTarget: string;
+
+	settings: CopyUrlInPreviewSettings;
+	async loadSettings() {
+		this.settings = Object.assign({}, DEFAULT_SETTINGS, await this.loadData());
+	}
+	async saveSettings() {
+		await this.saveData(this.settings);
+	}
+
+	async onload() {
+		await this.loadSettings();
+		this.addSettingTab(new CopyUrlInPreviewSettingTab(this.app, this));
+		this.registerDocument(document);
+		this.app.workspace.on("window-open", (_workspaceWindow, window) => {
+			this.registerDocument(window.document);
+		});
+		// register the image menu for canvas
+		this.registerEvent(
+			this.app.workspace.on("file-menu", (menu, file, source) => {
+				if (source === "canvas-menu"
+					&& file instanceof TFile
+					&& file.extension.match(/(avif|bmp|gif|jpe?g|png|svg|webp)/i)) {
+					menu.addItem((item) => item
+						.setIcon("image-file")
+						.setSection("system")
+						.setTitle(strings.menuItems.copyImageToClipboard)
+						.onClick(async () => {
+							const imageBuffer = await this.app.vault.readBinary(file);
+							const blob = new Blob([imageBuffer], { type: "image/png", });
+							try {
+								const data = new ClipboardItem({ [blob.type]: blob, });
+								await navigator.clipboard.write([data]);
+								new Notice(strings.messages.imageCopied, SUCCESS_NOTICE_TIMEOUT);
+							} catch (e) {
+								console.log(e);
+								new Notice(strings.messages.imageCopyFailed);
+							}
+						}));
+				}
+			})
+		);
+		this.registerEvent(
+			this.app.workspace.on("canvas:node-menu", (menu, node) => {
+				//@ts-ignore
```

Comment:
Please make an interface for this, like the ones in helpers.ts, instead of ts-ignore.